gym/envs/box2d/car_racing.py

```diff
@@ -438,235 +436,209 @@ def step(self, action):
-        for geom in self.viewer.onetime_geoms:
-            geom.render()
-        self.viewer.onetime_geoms = []
-        t.disable()
+        self.screen.fill(0)
+        self.screen.blit(self.surf, (0, 0))
+
+        # showing stats
         self.render_indicators(WINDOW_W, WINDOW_H)
 
-        if mode == "human":
-            win.flip()
-            return self.viewer.isopen
+        font = pygame.font.Font(pygame.font.get_default_font(), 42)
+        text = font.render("%04i" % self.reward, True, (255, 255, 255), (0, 0, 0))
+        text_rect = text.get_rect()
+        text_rect.center = (60, WINDOW_H - WINDOW_H * 2.5 / 40.0)
+        self.screen.blit(text, text_rect)
 
-        image_data = (
-            pyglet.image.get_buffer_manager().get_color_buffer().get_image_data()
-        )
-        arr = np.fromstring(image_data.get_data(), dtype=np.uint8, sep="")
-        arr = arr.reshape(VP_H, VP_W, 4)
-        arr = arr[::-1, :, 0:3]
-
-        return arr
+        if mode == "human":
+            pygame.display.flip()
 
-    def close(self):
-        if self.viewer is not None:
-            self.viewer.close()
-            self.viewer = None
-
-    def render_road(self):
-        colors = [0.4, 0.8, 0.4, 1.0] * 4
-        polygons_ = [
-            +PLAYFIELD,
-            +PLAYFIELD,
-            0,
-            +PLAYFIELD,
-            -PLAYFIELD,
-            0,
-            -PLAYFIELD,
-            -PLAYFIELD,
-            0,
-            -PLAYFIELD,
-            +PLAYFIELD,
-            0,
+        if mode == "rgb_array":
+            scaled_screen = pygame.transform.smoothscale(
+                self.screen, (VIDEO_W, VIDEO_H)
+            )
+            return np.transpose(
+                np.array(pygame.surfarray.pixels3d(scaled_screen)), axes=(1, 0, 2)
+            )
+        elif mode == "state_pixels":
+            scaled_screen = pygame.transform.smoothscale(
+                self.screen, (STATE_W, STATE_H)
+            )
+            return np.transpose(
+                np.array(pygame.surfarray.pixels3d(scaled_screen)), axes=(1, 0, 2)
+            )
+        else:
+            return self.isopen
+
+    def render_road(self, zoom, translation, angle):
+        bounds = PLAYFIELD
+        field = [
+            (2 * bounds, 2 * bounds),
+            (2 * bounds, 0),
+            (0, 0),
+            (0, 2 * bounds),
         ]
-
-        k = PLAYFIELD / 20.0
-        colors.extend([0.4, 0.9, 0.4, 1.0] * 4 * 20 * 20)
-        for x in range(-20, 20, 2):
-            for y in range(-20, 20, 2):
-                polygons_.extend(
+        trans_field = []
+        for coord in field:
+            coord = pygame.math.Vector2(coord).rotate_rad(angle)
+            coord = (coord[0] * zoom + translation[0], coord[1] * zoom + translation[1])
+            trans_field.append(coord)
+        gfxdraw.aapolygon(self.surf, trans_field, (102, 204, 102))
+        gfxdraw.filled_polygon(self.surf, trans_field, (102, 204, 102))
```

Comment:
See comment below about refactoring. This might also benefit from it, since I see there's some duplication.